packages/bloc/lib/src/bloc.dart

```diff
@@ -6,26 +6,53 @@ import 'package:rxdart/rxdart.dart';
 
 /// Takes a [Stream] of [Event]s as input
 /// and transforms them into a [Stream] of [State]s as output.
-abstract class Bloc<Event, State> {
+abstract class Bloc<Event, State> extends Stream<State> {
   final PublishSubject<Event> _eventSubject = PublishSubject<Event>();
 
   BehaviorSubject<State> _stateSubject;
 
-  /// Returns [Stream] of [State]s.
-  /// Usually consumed by the presentation layer.
-  Stream<State> get state => _stateSubject.stream;
+  /// Returns the current [State] of the [Bloc].
+  State get state => _stateSubject.value;
 
   /// Returns the [State] before any [Event]s have been `dispatched`.
   State get initialState;
 
-  /// Returns the current [State] of the [Bloc].
-  State get currentState => _stateSubject.value;
+  /// Returns whether the `Stream<State>` is a broadcast stream.
+  bool get isBroadcast => _stateSubject.stream.isBroadcast;
```

Comment:
Can just use `_stateSubject.isBroadcast` as a shortcut. This will always return `true`, since Subjects are all broadcast streams